### ISSUE TITLE
[java] UnusedAssignment false positive when using assert

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#5159]
**Affects PMD Version:** 7.4.0

**Rule:** https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedassignment

**Description:**

The rule incorrectly assumes that `assert` statements will always be executed.

**Code Sample demonstrating the issue:**

```java
{
	boolean assertionsAreEnabled = false;
	//noinspection AssertWithSideEffects
	assert assertionsAreEnabled = true;
	if (assertionsAreEnabled)
		return newInstance(value, name, getAssumeThatConfiguration());
	return MapValidatorNoOp.getInstance();
}
```

**Expected outcome:**

PMD reports a violation at line `boolean assertionsAreEnabled = false;` but that's wrong because if `assert` is not executed then the assignment is not overwritten.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Maven 3.24.0
